tests/testsuite/vendor.rs

```diff
@@ -631,3 +631,36 @@ fn config_instructions_works() {
         .with_stderr_contains("[..]foo/vendor/gitdep/src/lib.rs[..]")
         .run();
 }
+
+#[cargo_test]
+fn git_crlf_preservation() {
+    // Check that newlines don't get changed when you vendor
+    // (will only fail if your system is setup with core.autocrlf=true on windows)
```

Comment:
Can you update the test to actually set `core.autocrlf=true`?  You should be able to do that with `fs::write(paths::home().join(".gitconfig"), "…")`.

Reply:
Is the mocking sufficiently robust that this won't mess up the user's system?

Reply:
Yea, every test gets a synthetic home. [here](https://github.com/rust-lang/cargo/blob/f84f3f8c630c75a1ec01b818ff469d3496228c6b/tests/testsuite/git.rs#L2546-L2562) is an example.